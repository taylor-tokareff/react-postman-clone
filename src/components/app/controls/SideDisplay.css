.back {
  background-color: black;
}

.back {
  background-color: red;
}