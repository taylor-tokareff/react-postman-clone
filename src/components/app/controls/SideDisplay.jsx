import React from 'react';
import Side from './Side';
import { getMethod, getUrl } from '../UtilFunctions.js';
import styles from './SideDisplay.css';

const SideDisplay = () => {

  const method = getMethod();
  const url = getUrl();

  const sideElements = url.map((sideItem, i) => {
    return <li className={styles.back} key={sideItem}>
      <Side
        u={sideItem}
        m={method[i]}
      />
    </li>;
  });

  return (
    <ul className={styles.back} data-testid="side">
      {sideElements}
    </ul>
  );
};

export default SideDisplay;
