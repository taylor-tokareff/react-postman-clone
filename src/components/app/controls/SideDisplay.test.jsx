import React from 'react';
import { render } from '@testing-library/react';
import SideDisplay from './SideDisplay.jsx';

describe('sideDisplay component', () => {
  it('renders ul', () => {
    const { asFragment } = render(
      <SideDisplay />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
