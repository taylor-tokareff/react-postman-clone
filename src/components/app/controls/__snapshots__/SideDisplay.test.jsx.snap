// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sideDisplay component renders ul 1`] = `
<DocumentFragment>
  <ul
    class="back"
    data-testid="side"
  />
</DocumentFragment>
`;
